cpp/include/cudf/datetime.hpp

```diff
@@ -142,5 +142,19 @@ std::unique_ptr<cudf::column> extract_second(
     cudf::column_view const& column,
     rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
 
+/**
+ * @brief  Computes the last day of the month in date time type and returns a TIMESTAMP_DAYS
+ * cudf::column.
+ *
+ * @param[in] cudf::column_view of the input datetime values
+ *
+ * @returns cudf::column containing last day of the month as TIMESTAMP_DAYS
+ * @throw cudf::logic_error if input column datatype is not TIMESTAMP
+ */
+
+std::unique_ptr<cudf::column> last_day(
```

Comment:
yes, the requirement is from spark that wants to support `last_day` sql function

Reply:
I agree with Conor... even though there is a SQL function called "last_day", I think it is ambiguous. It could mean "last day of the year".  We don't have to name our functions the same as SQL if we can improve on it.

In fact, C++ std::chrono calls this `month_day_last`. https://en.cppreference.com/w/cpp/chrono/month_day_last

I prefer that we try to be consistent with C++ in our C++ API where we can agree that the C++ naming is "good".

Reply:
changed to `last_day_of_month`